src/components/shared/form/sign-up.tsx

```diff
@@ -0,0 +1,228 @@
+import { Link, navigate } from 'gatsby';
+import * as React from 'react';
+
+import styled from '@styled-components';
+import {
+  FormLabel,
+  FormInput,
+  LinkWrapper,
+  ButtonWrapper,
+} from '@components/shared/form';
+import Form from '@components/shared/form';
+import { useState } from 'react';
+import ServiceResolver from '@/api/service-resolver';
+import { ApiResponse, ErrorResponse } from '@/api/types/responses';
+import { UserValidation } from '@/api/types/user-validation';
+import { Button } from '@components/app/shared';
+import { JwtToken } from '@/api/types/jwt-token';
+import { SessionStorageHelper, UserAuthHelper } from '@/helpers';
+import { MockAuthService } from '@/mocks/mock-auth-service';
+import { AuthService } from '@/api/auth-service';
+import { ProjectUser } from '@/api/types/project-user';
+import { FormVal } from '@/utils/form-validation';
+
+const Wrapper = styled.section`
+  background-color: ${({ theme }) => theme.colors.section};
+
+  padding: ${({ theme }) => theme.boxes.padding.section.smallTop};
+
+  @media screen and(max-width: ${({ theme }) => theme.sizes.width.medium}) {
+    flex-direction: column;
+  }
+
+  @media screen and(max-width: ${({ theme }) => theme.sizes.width.small}) {
+    padding: ${({ theme }) => theme.boxes.padding.section.small};
+  }
+`;
+
+const Error = styled.ul`
+  color: ${({ theme }) => theme.colors.alert.danger};
+  margin: 0;
+
+  li {
+    margin: 0;
+    list-style: none;
+  }
+`;
+
+const UsernameCheck = styled.small<{ isValid: boolean }>`
+  color: ${(props) => (props.isValid ? '' : 'red')};
+`;
+
+export const SignUpForm: React.FC = () => {
+  const validation = new FormVal();
+
+  const [formInputs, setFormInputs] = useState({
+    email: { val: '', required: true },
+    username: { val: '', required: true },
+    password: { val: '', required: true },
+    confirmPassword: { val: '', required: true },
+  });
+
+  const [formErrors, setFormErrors] = useState<string[]>([]);
+  const [message, setMessage] = useState<string | UserValidation>();
+  const [auth, setAuth] = useState<MockAuthService | AuthService>();
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [usernameAvailablity, setUsernameAvailability] = useState<
+    UserValidation
+  >({ valid: false, reason: '' });
+
+  React.useEffect(() => {
+    setAuth(new ServiceResolver().AuthResolver());
```

Comment:
Consider this addition - 
```suggestion
    setApi(new ServiceResolver().ApiResolver());
    setAuth(new ServiceResolver().AuthResolver());
```

Reply:
@aaron2212 could you comment on this also?

Reply:
@roymoran Will, this work, since setApi expects either `MockAuthService` or `ApiService`, but `ApiResolver` only returns either `MockApiService` or `ApiService`?